proxy/proxy/httpserver.py

```diff
@@ -0,0 +1,171 @@
+        except TimeoutError as e:
+            # a read or a write timed out.  Discard this connection
+            self.log_error("Request timed out: %r", e)
+            self.close_connection = (  # pylint: disable=attribute-defined-outside-init
+                True
+            )
+            return
+
+    def do_POST(self):  # pylint: disable=invalid-name
+        """do POST"""
+        logging.debug("POST")
+        url = f"{dst_protocol}://" + self.headers["Host"] + self.path
+        logging.debug("Passthrough POST request: %s", url)
+        content_length = int(self.headers["Content-Length"])
+        data = self.rfile.read(content_length)
+        job_request = self.path.find("/runtime/jobs") != -1
+        middleware_job_id = None
+        if job_request:
+            if "X-Qx-Client-Application" in self.headers:
+                logging.debug(
+                    "X-Qx-Client-Application: %s",
+                    self.headers["X-Qx-Client-Application"],
+                )
+                pos = self.headers["X-Qx-Client-Application"].find("middleware_job_id")
+                if pos != -1:
+                    middleware_job_id = self.headers["X-Qx-Client-Application"][
+                        pos
+                        + len("middleware_job_id/") : pos
+                        + len("middleware_job_id/")
+                        + 36
+                    ]
+                    logging.debug("Middleware Job ID: %s", middleware_job_id)
+
+        resp = None
+        while resp is None:
+            try:
+                resp = requests.request(
+                    self.command, url, headers=self.headers, data=data, timeout=TIMEOUT
+                )
+            except:  # pylint: disable=bare-except
+                logging.debug("request error retrying")
+
+        self.handle_response(resp)
+        self.wfile.flush()  # actually send the response if not already done.
+        if job_request and middleware_job_id:
+            jsondata = json.loads(resp.content.decode("utf-8"))
+            logging.debug("job id: %s", jsondata["id"])
+
+            token = "awesome_token"
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {token}",
+            }
+            r = requests.post(
+                f"http://{gateway_url}/api/v1/jobs/{middleware_job_id}/add_runtimejob/",
+                headers=headers,
+                json={"runtime_job": jsondata["id"]},
+                timeout=TIMEOUT,
+            )
+            logging.debug("Gateway API Response: %s", r)
+        logging.debug("response from backend: %s", resp.status_code)
+
+
+if __name__ == "__main__":
+    server_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    server_context.load_cert_chain(
+        certfile="/etc/ray/tls/tls.crt", keyfile="/etc/ray/tls/tls.key"
+    )
+    server_context.load_verify_locations(cafile="/etc/ca/tls/ca.crt")
+    server_context.verify_mode = ssl.CERT_NONE
```

Comment:
This is totally outside my knowledge so just asking. We are configuring certificates for the proxy but configuring the mode as: `ssl.CERT_NONE`. Should not we use another mode?

Reply:
It may not have any effects because it is set at the server side.  I'll try if it works without setting it.

Reply:
Or if another mode has more sense. I don't really know here 😂